### PR TITLE
Math Magicians components

### PR DESCRIPTION
**Pull request by dazehinn**
### Features in this PR

- [ ] Deleted all the boilerplate from Create React App (text, images, styles).

- [ ] Created a new directory called components.

- [ ] Created a Calculator component to match UI

- [ ] Use functional components

- [ ] Created at least 1 child component

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import './calculator.css';
+
+const Calculator = () => (
```
Comment (Shedrack-Sunday) [inline src/components/calculator.js:5]: - [ ] It is important to import React from 'react' in this code.

- JSX, which is used in this code to define the structure of the Calculator component, requires React to be in scope. Without importing React, the code would throw possible   `ReferenceError` when attempting to use JSX.

- Therefore, the code should also have an import statement like this:
`import React from 'react';
`

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,10 @@
-import logo from './logo.svg';
 import './App.css';
+import Calculator from './components/calculator';
```
Comment (Shedrack-Sunday) [inline src/App.js]: - [ ] It is important to import ``React` from 'react' in this code.

- JSX, which is used in this code to define the structure of the component, requires React to be in scope. Without importing React, the code would throw possible ReferenceError when attempting to use JSX.

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @dazehinn      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights 
- Nice project setup✔️  
- Nice use Functional component✔️  
  
# Required Changes  🌵 
- [ ] You're missing a major import in your code.

_Check the comments for detailed review 



# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).


https://github.com/microverseinc/readme-template

File: src/components/calculator.js (hunk 1 above)
Comment (dazehinn) [reply src/components/calculator.js:5]: Hello @Shedrack-Sunday 
Thanks for your review. I actually had doubts whether to include it or not but when it showed no linter error, I thought it was ok.

File: src/App.js (hunk 2 above)
Comment (dazehinn) [reply src/App.js]: Does that mean for each React component that I create I should always import React from' react'?

Comment (Gambit142) [reply src/App.js]: @dazehinn, You no longer need to `import React from "react"`. Starting from the release of **_React 17_** of React, JSX is automatically transformed without using `React.createElement`. So if you are not using **_React 17_**, you need to but if you are using **_React 17_**, you don't need to. 👍

File: src/App.js [hunk 3]
```diff
@@ -1,25 +1,11 @@
-import logo from './logo.svg';
+import React from 'react';
 import './App.css';
```
Comment (Gambit142) [inline src/App.js:2]: - [ ] The import statement is wrong. It should be `'react'` instead of `'reaact'`. Kindly correct the typographical error. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @dazehinn 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- All linters checks passed ✔️ 
- Created child component ✔️ 
- Created Calculator component ✔️ 
- UI matches the design template provided ✔️ 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 3 above)
Comment (dazehinn) [reply src/App.js:2]: I see 'react instead of the 'reaact' which you talk about

File: src/App.js (hunk 2 above)
Comment (dazehinn) [reply src/App.js]: @Gambit142, this is very enlightening. Thanks. I'm actuallly using React 18.

File: src/App.js (hunk 3 above)
Comment (dazehinn) [reply src/App.js:2]: @Gambit142, based on your comment above I think I will just remove the import react statement completely.

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @dazehinn 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**